packages/dart/noports_core/lib/src/commands/activate/activate_impl.dart

```
@@ -0,0 +1,146 @@
+  /// Optionally uses [_params.atKeysFilePath] if provided.
+  ///
+  /// Returns: [AtEnrollmentResponse] containing enrollment status and details
+  /// Throws: [ArgumentError] if otp is missing
+  Future<int> enroll() async {
+    if (_params.otp == null) {
+      throw ArgumentError('Cannot create enrollment without otp');
+    }
+
+    logger.info(
+      'Creating new enrollment with deviceName: ${_params.deviceName}',
+    );
+
+    await _validateAndPrepareKeysFile();
+
+    final response = await _onboardingService.enroll(
+      _params.appName,
+      _params.deviceName!,
+      _params.otp!,
+      _params.namespaces,
+      atKeysFile: _getKeysFile(),
+    );
+
+    return response.enrollStatus == EnrollmentStatus.approved ? 0 : 1;
+  }
+
+  /// Validates and prepares the atKeys file location before enrollment.
+  ///
+  /// This method checks if a keys file already exists at the target location
+  /// (either specified by the user or at the default path) and prompts the user
+  /// to provide an alternate location if a file is found. This prevents
+  /// accidentally overwriting existing authentication keys during enrollment.
+  ///
+  /// The validation flow:
+  /// 1. If [_params.atKeysFilePath] is null, checks the default keys file path
+  ///    from the [AtOnboardingPreference]
+  /// 2. If [_params.atKeysFilePath] is provided, checks that custom path
+  /// 3. If a file exists at either location, prompts the user for an alternate
+  ///    path and updates [_params.atKeysFilePath]
+  /// 4. If no file exists, returns without modification (safe to proceed)
+  ///
+  ///
+  /// TODO: Replace [AtOnboardingServiceImpl] after ensuring that
+  /// [AtOnboardingService] throws a specific exception when keyFile exists at defaultPath
+  Future<void> _validateAndPrepareKeysFile() async {
+    final String? keysFilePath;
+
+    if (_params.atKeysFilePath == null) {
+      // Use the default path from onboarding service preferences
```

Workaround to avoid conflicts with creating atKeys files. Currently fetches the defaultPath from OnboardingServiceImpl.

Changes required: OnboardingService should throw a dedicated exception for key file name collisions, which can then be caught and handled.

---

Need to address this issue fully in this PR. Find the places that MUST have file existence checking (onboard, enroll), ensure the checking always happens there. Remove file existence checking from elsewhere and verify that all the ways we create atKeys files do end up having a file existence check so we don't ever overwrite keys files. As a final safeguard, put a check into the place we finally write an AtKeys file and do not permit overwriting.